debezium-connector-mysql/src/main/java/io/debezium/connector/mysql/SnapshotReader.java

```diff
@@ -122,8 +122,7 @@ protected void doInitialize() {
      */
     @Override
     protected void doStart() {
-        thread = new Thread(this::execute, "mysql-snapshot-" + context.serverName());
-        thread.start();
+        Threads.newSingleThreadExecutor(MySqlConnector.class, context.serverName(), "mysql-snapshot").execute(this::execute);
```

Comment:
I think "snapshot" instead of "mysql-snapshot" is better. The DB type already is part of the connector name.

Reply:
This leaks the executor (thread). It should be stopped upon shutdown.